bitcoin/src/network/constants.rs

```diff
@@ -150,6 +185,56 @@ impl FromStr for Network {
     }
 }
 
+/// Error in parsing magic from string.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct UnknownMagic(Magic);
+
+impl TryFrom<Magic> for Network {
+    type Error = UnknownMagic;
+
+    fn try_from(magic: Magic) -> Result<Self, Self::Error> {
+        match magic {
+            // Note: any new network entries must be added below
+            Magic::BITCOIN => Ok(Network::Bitcoin),
+            Magic::TESTNET => Ok(Network::Testnet),
+            Magic::SIGNET => Ok(Network::Signet),
+            Magic::REGTEST => Ok(Network::Regtest),
+            _ => Err(UnknownMagic(magic))
+        }
+    }
+}
+
+impl fmt::Display for Magic {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        write!(f, "{:#010x}", u32::from_be_bytes(self.0))
```

Comment:
Why does this have 0x prefix? Maybe calling `LowerHex::fmt(&u32::from_be_bytes(self.0), f)` would be better as it'd give people the opportunity to customize printing.

Reply:
How about implementing upper/lower hex traits for the type?

Reply:
Oh, yes, they should be provided. I just realized I made a mistake forwarding to `u32` is wrong because for magic with leading zero bytes those won't be printed.

Reply:
Cool, will do it in the follow up PR.

Reply:
This should work and support formatting logic out of the box:

```rust
bitcoin_internals::fmt_hex_exact!(f, 4, &self.0, bitcoin_internals::hex::Case::Lower)
```

If it doesn't feel free to ping me.